lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -286,6 +286,9 @@ defmodule Bamboo.SMTPAdapter do
     Enum.reduce(config, [], &to_gen_smtp_server_config/2)
   end
 
+  defp to_gen_smtp_server_config({:server, {:system, var}}, config) do
+    [{:relay, System.get_env(var)} | config]
```

Comment:
it could be easier to make it generic like this
```elixir
defp to_gen_smtp_server_config({conf, {:system, var}}, config) do
  to_gen_smtp_server_config({conf, System.get_env(var)}, config)
end
```
and then leave handling different types of configs in specific config for example tls requires atom and System.get_env returns string (this is actually why I realized it wasn't this easy)